circle.yml

```diff
@@ -1,18 +1,71 @@
-dependencies: 
-  cache_directories: 
+# Setup in CircleCI account the following ENV variables:
+# BINTRAY_ACCOUNT
+# BINTRAY_API_KEY
+# DOCKER_USER
+# DOCKER_EMAIL
+# DOCKER_PASSWORD
+general:
+  branches:
+#    only:
+#      - master
+#      - /v[0-9]+(\.[0-9]+)*/
+  artifacts:
+    - /tmp/st2-packages
+
+machine:
+  # Overwrite these ENV variables in parametrized (manual/API) builds
+  environment:
+    ST2_GITURL: https://github.com/StackStorm/st2
+    ST2_GITREV: master
+    ST2_DOCKERFILES_REPO: https://github.com/armab/st2-dockerfiles
+    BUILD_DOCKER: 0
+    DEPLOY_DOCKER: 0
+    DEPLOY_PACKAGES: 0
+  pre:
+    # Paswordless SSH between parallel build nodes
+    - sudo cp /home/ubuntu/.ssh/config /root/.ssh/config
+    # Need latest Docker version for some features to work (CircleCI by default works with outdated version)
+    - |
+      sudo curl -L -o /usr/bin/docker 'http://s3-external-1.amazonaws.com/circle-downloads/docker-1.9.0-circleci'
+      sudo chmod 0755 /usr/bin/docker
+      sudo service docker start
+
+checkout:
+  post:
+    - .circle/vars.sh
+
+dependencies:
+  cache_directories:
     - ~/.cache/pip
     - ~/wheelhouse
-  post: 
-    - .circle/services.sh
-    - ".circle/compose.sh run wheezy /bin/true"
-    - ".circle/compose.sh run wheezy"
-  pre: 
-    - .circle/cache-permissions.sh
-
-machine: 
-  services: 
-    - docker
+  pre:
+    - sudo .circle/configure-services.sh
+    - sudo .circle/fix-cache-permissions.sh
+    - sudo pip install wheel docker-compose
+    - docker-compose version
+  override:
+    - .circle/docker-compose.sh pull ${DISTRO}:
+        parallel: true
+  post:
+    - .circle/docker-compose.sh build ${DISTRO}:
+        parallel: true
 
 test:
   override:
-    - ".circle/compose.sh run wheezy bash -c 'cp /root/Gemfile* ./ && bundle exec rspec'"
+    - .circle/docker-compose.sh test ${DISTRO}:
+        parallel: true
+  post:
+    - .circle/docker.sh build st2bundle
+    - .circle/docker.sh build st2actionrunner st2api st2auth st2exporter st2notifier st2resultstracker st2rulesengine st2sensorcontainer
+    - .circle/docker.sh run st2api
+    - .circle/docker.sh test st2api 'st2 --version'
+
+deployment:
+  bintray:
```

Comment:
bintray `deployment` section as-is for now/disabled.
We'll need to decide about repository naming first, but that's later.

At least I have everything synced.